src/org/smssecure/smssecure/util/dualsim/SubscriptionManagerCompat.java

```diff
@@ -33,6 +34,14 @@ public SubscriptionManagerCompat(Context context) {
     }
   }
 
+  public static Optional<Integer> getDefaultMessagingSubscriptionId() {
+    if (Build.VERSION.SDK_INT < 22) {
+      return Optional.absent();
+    }
+
+    return SmsManager.getDefaultSmsSubscriptionId() < 0 ? Optional.<Integer>absent() : Optional.of(SmsManager.getDefaultSmsSubscriptionId());
```

Comment:
please break into a non-ternary if stmt.

Also, can the `Optional.<Integer>absent()` be replaced with `Optional.absent()` (like the build version guard block)